tools/peptidegenomiccoordinate/peptideGenomicCoodinate.py

```diff
@@ -0,0 +1,150 @@
+
+# 
+# Author: Praveen Kumar
+# University of Minnesota
+#
+# Get peptide's genomic coordinate from the protein's genomic mapping sqlite file (which is derived from the https://toolshed.g2.bx.psu.edu/view/galaxyp/translate_bed/038ecf54cbec)
+# 
+# python peptideGenomicCoordinate.py <peptide_list> <mz_to_sqlite DB> <genomic mapping file DB> <output.bed>
+# 
+#
+
+def main():
+    import sys
+    import sqlite3
+    conn = sqlite3.connect(sys.argv[2])
+    c = conn.cursor()
+    c.execute("DROP table if exists novel")
+    conn.commit()
+    c.execute("CREATE TABLE novel(peptide text)")
+    pepfile = open(sys.argv[1],"r")
+
+    for seq in pepfile.readlines():
+        seq = seq.strip()
+        c.execute('INSERT INTO novel VALUES ("'+seq+'")')
+        conn.commit()
+
+    c.execute("SELECT distinct psm.sequence, ps.id, ps.sequence from db_sequence ps, psm_entries psm, novel n, proteins_by_peptide pbp where psm.sequence = n.peptide and pbp.peptide_ref = psm.id and pbp.id = ps.id")
```

Comment:
If you novel table is not too large you could simply ask `psm.sequence in `
https://stackoverflow.com/questions/5766230/select-from-sqlite-table-where-rowid-in-list-using-python-sqlite3-db-api-2-0

Not sure if this is way faster, but you do not need to manipulate the sqlite table, which is not a good idea. As you do thins in-place currently.